.github/workflows/build_and_test.yml

```diff
@@ -447,7 +448,7 @@ jobs:
           export SKIP_PACKAGING=false
           echo "Python Packaging Tests Enabled!"
         fi
-        ./dev/run-tests --parallelism 1 --modules "$MODULES_TO_TEST"
+        ./dev/run-tests --parallelism 1 --modules "$MODULES_TO_TEST" --python-executables "$PYTHON"
```

Comment:
We have some places here referencing `python3.9` to show the installed packages, and would need to fix https://github.com/apache/spark/blob/master/dev/infra/Dockerfile together so the corresponding dependencies are available in the docker image too.

Reply:
Yes, that's the reason why this PR doesn't add Python 3.10/3.11/3.12. Currently, this PR only reusing the existing infra Dockerfile.